### ISSUE TITLE
Support more browsers in Meta-Gem

### ISSUE THREAD

**Issue by nicolai86** [johnbintz/jasmine-headless-webkit#67]
Hey,

is there any chance you're going to release a jasmine-headless gem, which could drive firefox/ webkit/ opera in headless mode?

Would be a great win for cross-browser testing...!

That being said, I love this gem! Great work ;)

**Comment by johnbintz** [comment]
Thanks for the kind words! While this would be an interesting idea, the Selenium & Jasmine folks pretty much have that covered. You can use the `jasmine:ci` command contained within the stock Jasmine rake tasks to run your specs on whatever browser you wish, using Selenium as the bridge. You can even do that headless with Xvfb, just like you would with JHW. With some new plugins coming out, you can even take care of CoffeeScript compilation and Rails 3.1 asset pipeline exports.

This gem is much more geared toward fast continuous testing running in a browser environment -- just enough to get jQuery and Backbone and such working, and working quickly, without needing to futz around with fake DOMs in standalone JS products like Node.js or Rhino.

I would gently suggest that, if you need to do true cross-browser tests for a Web app, that you look at doing acceptance tests instead of just unit tests. Using a tool like Cucumber, you can ensure that the actual usage of your app in each environment is correct. Hopefully, any of the breakage points in your code would be at the points of integration with the DOM, and unit tests probably aren't the most appropriate place for this. jQuery and similar libraries smooth over those rough edges, which is why I don't worry (as much!) when I'm just doing CT with `jasmine-headless-webkit`.

Hope that makes sense. I should add a note to the docs about using Jasmine's Selenium solution for true cross-browser unit testing.

**Comment by nicolai86** [comment]
Thanks for the `jasmine:ci` + Xvfb direction. Do you happen to have any good resources to get me started?

I'm currently using `jasmine-headless-webkit` for TDD on coffee-fied jQuery plugins - cucumber feels like overkill for a small jQuery plugin. I'd like to make sure the plugin works in most open source browsers - and again, setting up Selenium feels like an overkill. That's what I love about this gem - there's barely any setup required!

**Comment by johnbintz** [comment]
Yeah, Cucumber is overkill for a jQuery plugin, I agree. Unfortunately, I've never used `jasmine:ci` extensively, beyond just running `jasmine:ci` and going "Oh, that's nice." Looking through the source, it looks like you could use a shell script calling `rake` tasks or hook up a script like this to test everything against an already running Jasmine server:

``` ruby
require 'jasmine'
require 'jasmine/selenium_driver'
results = Hash[%w{firefox chrome safari ie}.collect do |browser|
  driver = Jasmine::SeleniumDriver.new(browser, 'http://localhost:8888/')
  driver.connect
  result = driver.run
  driver.disconnect
  [ browser, result ]
end]
```

Capturing `$stdout` within the call to `#connect` would allow you to get the results of the runs.

...actually, I was talking on here with another user who wanted to be able to run the resulting HTML files generated by JHW standalone. One could use the output of that, served by Rack (maybe), within this Selenium setup, while getting really nice console output. Hrmm... _ponders_

**Comment by johnbintz** [comment]
...so I set up a server-based mode in another branch of JHW and have been thinking about this again. I've been doing a bunch with modern Selenium within Capybara and am starting to feel comfortable with trying to hook up Selenium to the running server (then finding some way to get the data out to...something...to do console reporting and file writing and such). Once the server aspect of the code of solid, I'll start digging into this a bit more.
